nexus-sdk/src/main/java/io/nexusrpc/handler/OperationContext.java

```diff
@@ -68,21 +75,51 @@ public boolean isMethodCancelled() {
    * Add a listener for method cancellation. This will be invoked immediately before this function
    * returns if the method is already cancelled. The listener must not block. This is not reentrant
    * and therefore must not be called in another cancellation listener.
+   *
+   * @return this
    */
-  public void addMethodCancellationListener(OperationMethodCancellationListener listener) {
+  public OperationContext addMethodCancellationListener(
+      OperationMethodCancellationListener listener) {
     if (methodCanceller != null) {
       methodCanceller.addListener(listener);
     }
+    return this;
+  }
+
+  /**
+   * Associates links with the current operation to be propagated back to the caller. Links will
+   * only be attached on successful responses.
+   *
+   * @return this
+   */
+  public OperationContext addLinks(Link... links) {
+    this.links.addAll(Arrays.asList(links));
+    return this;
+  }
+
+  /**
+   * Associates links with the current operation to be propagated back to the caller. It is
+   * recommended to use {@link #addLinks(Link...)} to avoid accidental override. Links will only be
+   * attached on successful responses.
+   *
+   * @return this
+   */
+  public OperationContext setLinks(Link... links) {
```

Comment:
```suggestion
  public OperationContext setLinks(List<Link> links) {
```

IMO a good Java setter on these kinds of objects has a parameter to match its getter. Then again a good Java setter also returns void and we're not doing that either. I won't make this blocking.